File: src/main/java/net/jeikobu/jbase/Localized.java
package net.jeikobu.jbase;

import java.util.Locale;
import java.util.ResourceBundle;
import java.util.regex.Pattern;

public abstract class Localized {
    protected final String getLocalized(Locale locale, String key) {
        ResourceBundle bundle = ResourceBundle.getBundle(this.getClass().getSimpleName(), locale);
        return bundle.getString(key);
    }

    protected final String getLocalized(Locale locale, String key, Object... elements) {
        String localized = getLocalized(locale, key);

        Pattern p = Pattern.compile("\\{\\}");
        for (Object element: elements) {
            localized = p.matcher(localized).replaceFirst(element.toString());
        }

        return localized;
    }
}

File: src/main/java/net/jeikobu/kotomi/base/Localized.kt
package net.jeikobu.kotomi.base

import java.util.*
import java.util.regex.Pattern

abstract class Localized {
    fun getLocalized(locale: Locale, key: String): String {
        val bundle = ResourceBundle.getBundle(this.javaClass.simpleName, locale)
        return bundle.getString(key)
    }

    fun getLocalized(locale: Locale, key: String, vararg elements: Any): String {
        var localized = getLocalized(locale, key)

        val p = Pattern.compile("\\{\\}")
        for (element in elements) {
            localized = p.matcher(localized).replaceFirst(element.toString())
        }

        return localized
    }
}